janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -319,7 +319,7 @@ public void close() {
                     config.getVertexCacheSize(), effectiveVertexCacheSize, MIN_VERTEX_CACHE_SIZE);
         }
 
-        vertexCache = new GuavaVertexCache(effectiveVertexCacheSize,concurrencyLevel,config.getDirtyVertexSize());
+        vertexCache = new CaffeineVertexCache(effectiveVertexCacheSize,config.getDirtyVertexSize());
```

Comment:
Can you also remove `GuavaVertexCache` class since it's not being used anymore?

Reply:
As far as I understand it is used in the benchmark test class. I guess we could move that class into testing module then

Reply:
I guess we can remove that class with the benchmark test class in the follow up PRs but I'm OK either way.

Reply:
Should the `GuavaVertexCache` be moved there rather than have it linger around as dead code?

Reply:
Actually I think it's better to leave the benchmark test class in this PR for historical reasons. We can move `GuavaVertexCache` to test module and then later after the full transition to Caffeine cache is done we can remove those test classes with the benchmark classes. It's just good if those benchmark classes are in git history

Reply:
>  later after the full transition to Caffeine cache is done we can remove those test classes with the benchmark classes

That sounds good to me!

Reply:
I agree to leave GuavaVertexCache there. I'm ok to compose several following PRs to clean it up.
Meanwhile, I found there are some other Guava cache, like https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/subquerycache/GuavaSubqueryCache.java, and it is supposed to be replaced by Caffeine cache also, but I have not yet benchmark it.

Reply:
> I agree to leave GuavaVertexCache there. I'm ok to compose several following PRs to clean it up. Meanwhile, I found there are some other Guava cache, like https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/subquerycache/GuavaSubqueryCache.java, and it is supposed to be replaced by Caffeine cache also, but I have not yet benchmark it.

There are 8 caches in JanusGraph actually and all of them need to be replaced by Caffeine (not in the current PR but in follow up PRs). I posted those places here: https://github.com/JanusGraph/janusgraph/issues/871#issuecomment-1230207577